### DISCUSSION TITLE
Validations of strings input to `text`

### DISCUSSION THREAD

**Discussion by jamesbraza** [sqlalchemy/sqlalchemy#10395] [Usage Questions] [answered]
It's possible to put non-SQL string into `text`, which of course leads to failed queries.

Is there some way to validate a `text` has a valid SQL string inside, before running a query with it?

```python
from sqlalchemy import text

text_clause = text(command)
text_clause.validate()
cursor = connection.execute(statement=text_clause)
```

**Comment by zzzeek** [comment, marked as answer]
nothing in SQLAlchemy, every database has its own version of  SQL so the database itself is of course the best source of validation of the SQL.    beyond that you'd have to use something like [SQLParse](https://pypi.org/project/sqlparse/) though it says it's non validating...maybe something like https://pypi.org/project/sqlvalidator/

**Reply by jamesbraza** [reply]
Okay that makes sense, will look into those thank you and appreciated!
